lib/ansible/modules/windows/win_disk_management.ps1

```diff
@@ -0,0 +1,1411 @@
+#!powershell
+#
+# Copyright 2017, Marc Tschapek <marc.tschapek@itelligencegroup.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+# Define script environment variables
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2
+
+# Parse arguments
+$params = Parse-Args -arguments $args -supports_check_mode $true
+
+## Extract each attributes into a variable
+# Find attributes
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$Logging = Get-AnsibleParam -obj $params -name "logging" -type "str" -default "standard" -ValidateSet "standard","verbose"
+$Size = Get-AnsibleParam -obj $params -name "size" -type "str/int"
+$FindPartitionStyle = Get-AnsibleParam -obj $params -name "partition_style_select" -type "str" -default "raw" -ValidateSet "raw","mbr","gpt"
+$OperationalStatus = Get-AnsibleParam -obj $params -name "operational_status" -type "str" -default "offline" -ValidateSet "offline","online"
+$ReadOnly = Get-AnsibleParam -obj $params -name "read_only" -type "bool" -default $true
+$Number = Get-AnsibleParam -obj $params -name "number" -type "str/int"
+# Set attributes partition
+$SetPartitionStyle = Get-AnsibleParam -obj $params -name "partition_style_set" -type "str" -default "gpt" -ValidateSet "gpt","mbr"
+$DriveLetter = Get-AnsibleParam -obj $params -name "drive_letter" -type "str"
+# Set attributes file system
+$FileSystem = Get-AnsibleParam -obj $params -name "file_system" -type "str" -default "ntfs" -ValidateSet "ntfs","refs"
+$Label = Get-AnsibleParam -obj $params -name "label" -type "str" -default "ansible_disk"
+$AllocUnitSize = Get-AnsibleParam -obj $params -name "allocation_unit_size" -type "str/int" -default 4 -ValidateSet 4,8,16,32,64
+# Set attributes switches
+$LargeFRS = Get-AnsibleParam -obj $params -name "large_frs" -type "bool" -default $false
+$ShortNames = Get-AnsibleParam -obj $params -name "short_names" -type "bool" -default $false
+$IntegrityStreams = Get-AnsibleParam -obj $params -name "integrity_streams" -type "bool" -default $false
+
+# Create a new result object
+$result = @{
+    changed = $false
```

Comment:
TBH with you, I don't think we should be adding all these options to the result. I think we should just be returning  `changed` and nothing else. The amount of code that is in this module to handle all this is quite large and makes it seem more complex than what it is.

If you really want logging options I would create a function that writes the log information to a file on the Windows both like what `win_updates` does and just keep on appending log info with datetime stamps.

As for the disk information stuff, keep it out as it should really be part of a facts/stat module instead and just causes more complication here.

Reply:
removed a lot of logging, added diff_mode